### PR TITLE
Refactor entrypoint

### LINKED ISSUE #33: Encryption keys should be stored
The file `/var/www/html/application/config.security.php` contains 2 encryption keys that should be properly volumed, otherwise each encrypted data would be lost after a reboot.

### PR DESCRIPTION
**Pull request by martialblog**
Rewrote the entrypoint to be a bit more flexible. 

- Debugging can be adjusted
- Data encryption keys can now be provided (plus a warning if you use non-persistent keys) - Fixes #33 
- Should be easier to add new variables in the future

### CODE REVIEW

File: 3.0/apache/entrypoint.sh [hunk 1]
```diff
@@ -29,8 +45,7 @@ if [ -z "$DB_SOCK" ]; then
     done
 fi
 
-
-# Check if already provisioned
+# Check config if already provisioned
```
Comment (jacksgt) [inline 3.0/apache/entrypoint.sh:48]: "Check if config already provisioned"

File: 3.0/apache/entrypoint.sh [hunk 2]
```diff
@@ -39,60 +54,83 @@ else
...
+      'rules' => array(),
+      'showScriptName' => true,
+    ),
+  ),
+  'config'=>array(
+    'publicurl'=>'$PUBLIC_URL',
+    'debug'=>$DEBUG,
+    'debugsql'=>$DEBUG_SQL,
+  )
+);
```
Comment (jacksgt) [inline 3.0/apache/entrypoint.sh:99]: This is nice!

File: 3.0/apache/entrypoint.sh [hunk 3]
```diff
@@ -39,60 +54,83 @@ else
...
+
+EOF
+
 fi
 
+# Check security if already provisioned
```
Comment (jacksgt) [inline 3.0/apache/entrypoint.sh:105]: "Check if security config already provisioned"

File: README.md [hunk 4]
```diff
@@ -52,6 +52,16 @@ To change to LimeSurvey configuration, you can mount a Volume into the Container
 
 **Hint**: If this configuration is present before the installation, the LimeSurvey Web Installer will not run automatically.
 
+## Data Encryption
+
+LimeSurvey 4 supports data encryption, this image give you these options:
+
+* Provide a security.php file directly
```
Comment (jacksgt) [inline README.md:59]: ... (via volume mount)

File: README.md [hunk 5]
```diff
@@ -52,6 +52,16 @@ To change to LimeSurvey configuration, you can mount a Volume into the Container
...
+* Provide encryption keys for the security.php file
```
Comment (jacksgt) [inline README.md:60]: ... (via environent variables)

File: 3.0/apache/entrypoint.sh (hunk 1 above)
Comment (martialblog) [reply 3.0/apache/entrypoint.sh]: :facepalm: fixed

File: README.md (hunk 5 above)
Comment (martialblog) [reply README.md]: Done

File: 3.0/apache/entrypoint.sh (hunk 2 above)
Comment (martialblog) [reply 3.0/apache/entrypoint.sh:99]: :+1:

File: README.md (hunk 4 above)
Comment (martialblog) [reply README.md]: Done

File: README.md [hunk 6]
```diff
@@ -52,6 +52,16 @@ To change to LimeSurvey configuration, you can mount a Volume into the Container
 
 **Hint**: If this configuration is present before the installation, the LimeSurvey Web Installer will not run automatically.
 
+## Data Encryption
+
+LimeSurvey 4 supports data encryption, this image give you these options:
+
+* Provide a security.php file directly (environment variables)
+* Provide encryption keys for the security.php file (volume)
```
Comment (jacksgt) [inline README.md:60]: Did you accidentally switch the explanation of these two lines?

If I provide the security.php directly, then presumably it's via volume.
If I just provide the keys themselves, the presumably the image configures the security.php file.

Right?

Comment (martialblog) [reply README.md:60]: I shouldn't do any changes today :facepalm:

Comment (jacksgt) [reply README.md]: It's okay :D

**Pull request merged**